### DISCUSSION TITLE
Problem with importing SI-Units

### DISCUSSION THREAD

**Discussion by weigher12** [OpenModelica/OpenModelica#12766] [Q&A] [answered]
Hi,

I had a problem when I tried to import a SI unit into a model, in OpenModelica v1.23.1 (64-bit)

I used the code from Modelica by example - Importing physical types.
https://mbe.modelica.university/components/packages/nimport/

The error i got was:
[Class Temperature not found in scope test3.]

here's the code I used:

model test3
  import Modelica.SIunits.Temperature;  
  parameter Temperature T_inf=3; 
  Temperature T;
  equation
  T = T_inf +30 ;
end test3;

I looked into the structure of SI units in OMEdit and found that i have to write it differently.
Modelica.Units.SI.Temperature instead of Modelica.SIunits.Temperature;

with this change the model works just fine

model test3
  import Modelica.Units.SI.Temperature;  
  parameter Temperature T_inf=3; 
  Temperature T;
  equation
  T = T_inf +30 ;
end test3;

**Comment by AnHeuermann** [comment, marked as answer]
Modelica by example is a bit older than the latest main release of the Modelica standard library (MSL) v4.0.0. Previously MSL v3.2.3 had `Modelica.SIunits` and MSL 4.0.0 changed it to be in `Modelica.Units.SI`.

OpenModelica ships with both MSL 4.0.0 and 3.2.2 or you can install it via the library manager:
![grafik](https://github.com/user-attachments/assets/71c0a8c8-9d71-462b-8667-7b514a728351)

You have three options:
  1. You can try to update the examples to work with the never version of the MSL. You'll probably learn the most when trying to stick with the never version of the MSL.
  2. Unload `Modelica`, `ModelicaServices` and `Complex` and load `Modelica` v3.2.3
  3. Change the Modelica version OMEdit loads by default to v3.2.3:
      OMEdit->Tools->Options->Libraries. Uncheck `Load latest Modelica version on startup` and `Add` library `Modelica` version `3.2.3+maint.om`.
      ![grafik](https://github.com/user-attachments/assets/3603daae-798a-4f28-8655-061f6138e78a)
Don't forget to change it back after you are done with Modelica by example to use the latest and greatest. Version MSL 4.1. will be released soon (tm).

**Comment by casella** [comment]
I guess we should ask @mtiller if he has plans to update ModelicaByExample to MSL 4.x.x. I think it would only be a minor effor, and maybe we can help him with that.
